sql/analyzer/resolve_tables.go

```diff
@@ -126,8 +128,63 @@ func resolveTable(ctx *sql.Context, t *plan.UnresolvedTable, a *Analyzer) (sql.N
 		return handleTableLookupFailure(err, name, db, a, t)
 	}
 
+	resolvedTableNode := plan.NewResolvedTable(rt, database, nil)
+
+	// Check for the information_schema.columns table which needs to resolve all defaults
+	// TODO: Is it okay to put something one off like this?
+	if database.Name() == "information_schema" && rt.Name() == "columns" {
+		return handleInfoSchemaColumnsTable(ctx, resolvedTableNode, a.Catalog)
+	}
+
 	a.Log("table resolved: %s", t.Name())
-	return plan.NewResolvedTable(rt, database, nil), nil
+	return resolvedTableNode, nil
+}
+
+// handleInfoSchemaColumnsTable wraps the information_schema.columns with an information_schema.ColumnsNode
+// for additional analyzer work.
+func handleInfoSchemaColumnsTable(ctx *sql.Context, rt sql.Node, c sql.Catalog) (sql.Node, error) {
+	tableColumnsToDefaultValue, err := getAllColumnsWithADefaultValue(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+
+	return information_schema.CreateNewColumnsNode(rt, tableColumnsToDefaultValue), nil
+}
+
+// getAllColumnsWithADefaultValue returns a map of columns of column names to columns. Each column stored in this map
+// is a Column that has a non-nil default value.
+func getAllColumnsWithADefaultValue(ctx *sql.Context, catalog sql.Catalog) (map[string]*sql.Column, error) {
+	ret := make(map[string]*sql.Column)
+
+	for _, db := range catalog.AllDatabases(ctx) {
+		err := sql.DBTableIter(ctx, db, func(t sql.Table) (cont bool, err error) {
+			for _, col := range t.Schema() {
+				if col.Default == nil {
+					continue
+				}
+
+				if ucd, ok := col.Default.Expression.(sql.UnresolvedColumnDefault); ok {
```

Comment:
You don't need to do this here, UnresolvedColumnDefault will get resolved in resolve_column_defaults